### PR TITLE
[Function]Support array_position function

### PR DESCRIPTION
**Pull request by SaintBacchus**
Impl the `array_position` function based on the function `array_contains`.
These two functions are very similar, but the `array_position` return index which begain from 1 and `array_contains` return true or not.

### CODE REVIEW

**Comment by imay** [comment]
@SaintBacchus Thanks for your contribution.
IMO, I think `array_position` is a better name. Other system like prest/trino, spark, snowflake use this name.
So I suggest that use `array_position` as the new function name.

**Comment by SaintBacchus** [comment]
@imay Thanks for your suggestion. I will change the name of it.

File: be/src/exprs/vectorized/array_functions.cpp [hunk 1]
```diff
@@ -375,7 +378,7 @@ class ArrayContainsImpl {
         for (size_t i = 0; i < num_array; i++) {
             size_t offset = offsets_ptr[i];
             size_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
-            uint8_t found = 0;
+            size_t found = 0;
```
Comment (imay) [inline be/src/exprs/vectorized/array_functions.cpp:381]: It's better to name it position? And turn it to bool when assign it to result?

**Review by imay — COMMENTED** [review]
Great Code

File: be/src/exprs/vectorized/array_functions.cpp (hunk 1 above)
Comment (SaintBacchus) [reply be/src/exprs/vectorized/array_functions.cpp]: I add a new local var to indicate the postion. Will it be fine?

**Review by imay — APPROVED** [review]
LGTM

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
Need add a new doc for this the new function

**Comment by SaintBacchus** [comment]
OK, I will add it later

**Pull request merged**